File: solutions/cpp/Number Complement.cpp
#include <string>
#include <math.h>
using namespace std;
class Solution
{
public:
    int getBinaryNumber(string num)
    {
        int i = num.length() - 1;
        int sum = 0;
        int Pow = 0;
        while (i >= 0)
        {
            int n = num[i] - '0';
            sum = sum + n * pow(2, Pow);
            i = i - 1;
            Pow = Pow + 1;
        }
        return sum;
    }
    string getRemainder(int num)
    {
        if (num % 2 == 1)
            return "0";
        else
            return "1";
    }
    int findComplement(int num)
    {
        string bin_num = "";
        while (num > 0)
        {
            bin_num = getRemainder(num) + bin_num;

            num = num / 2;
        }
        return getBinaryNumber(bin_num);
    }
};

File: solutions/python/Number Complement.py
class Solution:
    def getBinaryNumber(self,bin_number:str)->int:
        sum=0
        i=len(bin_number)-1;
        Pow=0
        while(i>=0):
            sum=sum+(int(bin_number[i])*pow(2,Pow))
            Pow+=1
            i-=1;
        return sum;
    def findComplement(self, num: int) -> int:
        bin_num=""
        while(num>0):
            rem=num%2
            if rem==1:
                rem="0"
            if rem==0:
                rem="1"
            bin_num=rem+bin_num
            num=num//2;
        return self.getBinaryNumber(bin_num)
